cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/flow/FlowletProgramRunner.java

```diff
@@ -272,6 +276,23 @@ public ProgramController run(Program program, ProgramOptions options) {
                                    Lists.<ProcessSpecification<?>>newLinkedList());
       List<ConsumerSupplier<?>> consumerSuppliers = queueConsumerSupplierBuilder.build();
 
+      // make sure that the user has READ permission on all the streams
+      Principal principal = authenticationContext.getPrincipal();
+      for (ProcessSpecification processSpecification : processSpecs) {
```

Comment:
I think it's better do the check at the time when we try to create the `StreamQueueReader`. I think the checking can be done easily from the `QueueReaderFactory`, which is an injected field to `FlowletProgramRunner`. Since it is injected, it can take a `AuthorizationEnforcer`.